src/meta/processors/job/BalancePlan.cpp

```diff
@@ -92,36 +100,48 @@ void BalancePlan::invoke() {
           stopped = stopped_;
         }
         if (finished) {
-          CHECK_EQ(j, this->buckets_[i].size() - 1);
-          onFinished_();
-        } else if (j + 1 < this->buckets_[i].size()) {
-          auto& task = this->tasks_[this->buckets_[i][j + 1]];
+          CHECK_EQ(j, buckets_[i].size() - 1);
+          onFinished_(stopped ? meta::cpp2::JobStatus::STOPPED : meta::cpp2::JobStatus::FAILED);
+        } else if (j + 1 < buckets_[i].size()) {
+          auto& task = tasks_[buckets_[i][j + 1]];
```

Comment:
Since you have group by `spaceId + partId`, all tasks in a bucket should have same `spaceId + partId`, there should not be a `if` statement below?

BTW, you'd better add a unit test about this case.

Reply:
> Since you have group by `spaceId + partId`, all tasks in a bucket should have same `spaceId + partId`, there should not be a `if` statement below?
> 
> BTW, you'd better add a unit test about this case.

it has been removed